metadata-ingestion/src/datahub/ingestion/source/fs/fs_base.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass
+from typing import Iterable
+from urllib import parse
+from abc import ABCMeta, abstractmethod
+
+
+@dataclass
+class FileStatus:
+    path: str
+    size: int
+    is_file: bool
+
+    def __str__(self):
+        return f"FileStatus({self.path}, {self.size}, {self.is_file})"
+
+
+class FileSystem(metaclass=ABCMeta):
+
+    @classmethod
+    def create_fs(cls) -> "FileSystem":
```

Comment:
would be more consistent to just call this `create`

also should this method be taking kwargs?

Reply:
done